src/Silicon.cpp

```diff
@@ -536,136 +573,163 @@ namespace galsim {
 
         const int nphotons = photons.size();
 
-        GaussianDeviate gd(ud,0,1); // Random variable from Standard Normal dist.
-
         double addedFlux = 0.;
-        for (int i=0; i<nphotons; i++) {
-            // Update shapes every _nrecalc electrons
-            if (addedFlux > next_recalc) {
-                dbg<<"updatePixelDistortions because "<<addedFlux<<" > "<<next_recalc<<std::endl;
-                updatePixelDistortions(_delta.view());
-                target += _delta;
-                _delta.setZero();
-                next_recalc = addedFlux + _nrecalc;
-            }
-
-            // Get the location where the photon strikes the silicon:
-            double x0 = photons.getX(i); // in pixels
-            double y0 = photons.getY(i); // in pixels
-            xdbg<<"x0,y0 = "<<x0<<','<<y0;
-
-            double dz = calculateConversionDepth(photons, i, ud);
-            if (photons.hasAllocatedAngles()) {
-                double dxdz = photons.getDXDZ(i);
-                double dydz = photons.getDYDZ(i);
-                double dz_pixel = dz * invPixelSize;
-                x0 += dxdz * dz_pixel; // dx in pixels
-                y0 += dydz * dz_pixel; // dy in pixels
-            }
-            xdbg<<" => "<<x0<<','<<y0;
-            // This is the reverse of depth. zconv is how far above the substrate the e- converts.
-            double zconv = _sensorThickness - dz;
-            if (zconv < 0.0) continue; // Throw photon away if it hits the bottom
-            // TODO: Do something more realistic if it hits the bottom.
-
-            // Now we add in a displacement due to diffusion
-            if (_diffStep != 0.) {
-                double diffStep = std::max(
-                    0.0, diffStep_pixel_z * std::sqrt(zconv * _sensorThickness));
-                x0 += diffStep * gd();
-                y0 += diffStep * gd();
-            }
-            xdbg<<" => "<<x0<<','<<y0<<std::endl;
-            double flux = photons.getFlux(i);
+	std::vector<GaussianDeviate> gd;
+	for (int i = 0; i < numThreads; i++) {
+	   gd.push_back(GaussianDeviate(ud[i], 0, 1));
```

Comment:
Lots of places here where you use 3 spaces rather than our standard 4.  These should be fixed.